### PR TITLE
Add tox.ini file to implement rocks stanity tests

### PR DESCRIPTION
**Pull request by ca-scribner**
tox.ini and tests folders adapted from [oidc-authservice-rock](https://github.com/canonical/oidc-authservice-rock/).  

Part of #7

### CODE REVIEW

File: .github/workflows/on_push.yaml [hunk 1]
```diff
@@ -14,3 +14,8 @@ jobs:
     permissions:
       pull-requests: read
     secrets: inherit
+    with:
+      rockcraft-channel: latest/edge
```
Comment (orfeas-k) [inline .github/workflows/on_push.yaml:18]: Regarding rockcraft-channel, ~~if we don't use a specific feature from last `1.1.0`, then I think we could omit this input and go with `stable` here~~
I saw that they have released the fixes in `stable` (see [discussion in matrix](https://matrix.to/#/!NPPCseDHKRvSBMUEXN:ubuntu.com/$sX6hnpqkzKsWurwzUPcXHgLqA8gdOR1mCUi61jTibEc?via=ubuntu.com&via=matrix.org) or [snap](https://snapcraft.io/rockcraft))

File: .github/workflows/on_pull_request.yaml [hunk 2]
```diff
@@ -11,3 +11,8 @@ jobs:
     permissions:
       pull-requests: read
     secrets: inherit
+    with:
+      rockcraft-channel: latest/edge
```
Comment (orfeas-k) [inline .github/workflows/on_pull_request.yaml:15]: same with on_push

File: dex/tox.ini [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+[tox]
+skipsdist = True
+skip_missing_interpreters = True
+envlist = unit, sanity, integration
```
[the remark this answers is not in the document]
Comment (ca-scribner) [reply dex/tox.ini]: good call.  fixed

File: .github/workflows/on_push.yaml (hunk 1 above)
Comment (ca-scribner) [reply .github/workflows/on_push.yaml]: updated to stable.  I'd pin to a version, but they don't have versioned channels :(

File: .github/workflows/on_pull_request.yaml (hunk 2 above)
Comment (ca-scribner) [reply .github/workflows/on_pull_request.yaml]: updated to stable.  I'd pin to a version, but they don't have versioned channels :(

**Review by orfeas-k — APPROVED** [review]
LGTM!

**Pull request merged**